### DISCUSSION TITLE
Bug: Edges of blur radius are not blurred

### DISCUSSION THREAD

**Discussion by durkester** [chrisbanes/haze#447] [General]
The greater the blur radius, the less of the edge of the top haze state is blurred. For example, I'm using 40.dp so the hazeChild text is readable, but the blurring algorithm isn't accounting for edges within that 40.dp section. Or at least not fully.

**Comment by chrisbanes** [comment]
No code?  ✅
No screenshot? ✅

How do you expect anyone to reply to this?

FYI: the ‘blur algorithm’ very much **does** work up to edges, but it sounds like you’re not setting an appropriate background color.

**Reply by durkester** [reply]
Good point. I guess I figured it was a known issue or there was a work around I couldn't find.

```
val hazeState = remember { HazeState() }

    Box(

    ) {
        Image(
            painter = painterResource(id = R.drawable.map_light_full),
            contentDescription = "icon",
            contentScale = ContentScale.FillBounds,
            modifier = Modifier
                .fillMaxWidth(fraction = 1f)
                .fillMaxHeight(fraction = 1f)
                .haze(hazeState)
        )

        Box(
            Modifier
                .fillMaxSize()
                .hazeChild(state = hazeState) {
                    backgroundColor = Color.White.copy(alpha = 0f)
                    tints = listOf(
                        HazeTint(Color.White.copy(alpha = 0.0f))
                    )
                    blurRadius = 40.dp
                    noiseFactor = 0.0f
                }
        ) {

        }
    }
```

![Screenshot 2024-12-16 010513](https://github.com/user-attachments/assets/1a3d9034-551a-428b-9f45-8bed316c4657)
^Here is 80.dp blur radius with a transparent background to show that the edges are not blurred.

![Screenshot 2024-12-16 010601](https://github.com/user-attachments/assets/ec5a4ae4-91a4-4b72-b701-c041acf22a49)
^And 10.dp to show the difference.

![Screenshot 2024-12-16 010642](https://github.com/user-attachments/assets/787a0a8c-337d-431e-9a05-def9a4c322c1)
^Yes making the background white when the average blur is a light color works, but how do you handle it when the background color is dynamic and could be dark?
![Screenshot 2024-12-16 010742](https://github.com/user-attachments/assets/5c0662de-b81c-48a2-b4ef-dcf2a86cdf7b)

I tried looking if the HazeState exposes like an average color or something to set as the background but couldn't find anything.

And making the background black or any other color doesn't fix this issue.
![Screenshot 2024-12-16 010944](https://github.com/user-attachments/assets/b68529d2-66ae-49b3-8e4c-7f2239efc72c)
![Screenshot 2024-12-16 011002](https://github.com/user-attachments/assets/c3c4c7f0-a3c1-4693-9e98-2fc6cfe60a5e)
You get the edges in the background color.

Compared with my Honor Magic V3 which has a notification shade that blurs the background and has a nice even blur across the view.
![Screenshot_20241216_011142](https://github.com/user-attachments/assets/edee78b5-714a-407a-b015-2bb04e692b82)

I found a decent workaround of layering 2 hazeChilds
```
.hazeChild(state = hazeState1) {
    backgroundColor = gray
    tints = listOf(
        HazeTint(black.copy(alpha = 0.15f))
    )
    blurRadius = 20.dp
    noiseFactor = 0.0f
}
.hazeChild(state = hazeState1) {
    backgroundColor = Color.Transparent
    tints = listOf(
        HazeTint(black.copy(alpha = 0.15f))
    )
    blurRadius = 60.dp
    noiseFactor = 0.0f
}
```
Probably not great for performance but the top's background color pretty much doesn't matter as the lower blurRadius blurs the background enough to average the background color to blend in well with the bottom hazeChild.

**Reply by chrisbanes** [reply]
The answer is simple, don't use a transparent `backgroundColor`. A blurring algorithm can only work with pixels which have a value, and transparent pixels have no value.

**Reply by durkester** [reply]
![Screenshot 2024-12-16 012555](https://github.com/user-attachments/assets/27582fe3-4670-4c72-b49b-f87765ec1ad2)
So you have no answer to edges not matching?

**Reply by durkester** [reply]
These are solid colors

**Reply by chrisbanes** [reply]
"No answer" 🙄. Feel free to go and implement this yourself.
